src/tokenizer.ts

```diff
@@ -1485,12 +1480,50 @@ export class Tokenizer extends DiagnosticEmitter {
         ) {
           ++this.pos;
         }
-        while (this.pos < end && isDecimalDigit(text.charCodeAt(this.pos))) {
-          ++this.pos;
+        this.readDecimalFloatPartial();
+      }
+    }
+    let floatString = text.substring(start, this.pos).replaceAll("_", "");
```

Comment:
Also could you add `hasSep` boolean variable which set to true if at least one separator found and do `replaceAll` only for this case like:
```ts
  ...
  let result = text.substring(start, this.pos)
  if (hasSep) result = result.replaceAll("_", "");
  return parseFloat(result);
}
```